### PR TITLE
Add conflict resolution for files

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: pkg/sharing/files.go [hunk 1]
```diff
@@ -648,7 +651,10 @@ func (s *Sharing) CreateDir(inst *instance.Instance, target map[string]interface
 		return err
 	}
 	dir.SetID(target["_id"].(string))
+	ref.SID = consts.Files + "/" + dir.DocID
 	copySafeFieldsToDir(target, dir)
+	_, ruleIndex := s.findRuleForNewDirectory(dir)
```
Comment (Gara64) [inline pkg/sharing/files.go]: We might want to check if the rule is `ǹil` to make sure the ruleIndex is legit

File: pkg/sharing/upload.go [hunk 2]
```diff
@@ -317,6 +318,10 @@ func (s *Sharing) SyncFile(inst *instance.Instance, target *FileDocWithRevisions
 	if _, ok := ref.Infos[s.SID]; !ok {
 		return nil, ErrSafety
 	}
+	if ref.Revisions.Find(target.DocRev) != nil {
+		// It's just the echo, there is nothing to do
```
Comment (Gara64) [inline pkg/sharing/upload.go]: :)

**Comment by Gara64** [comment]
Well done with this PR! It's sometimes quite complex, so tests are definitely needed :)

File: pkg/sharing/files.go (hunk 1 above)
Comment (nono) [reply pkg/sharing/files.go]: Good catch!

**Pull request merged**